[2024-06-12T09:22:17.587+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-12T09:22:17.655+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_final_dag.spotify_etl_final scheduled__2024-06-12T07:10:00+00:00 [queued]>
[2024-06-12T09:22:17.688+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_final_dag.spotify_etl_final scheduled__2024-06-12T07:10:00+00:00 [queued]>
[2024-06-12T09:22:17.689+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-06-12T09:22:17.773+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): spotify_etl_final> on 2024-06-12 07:10:00+00:00
[2024-06-12T09:22:17.803+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=76) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-12T09:22:17.807+0000] {standard_task_runner.py:63} INFO - Started process 87 to run task
[2024-06-12T09:22:17.837+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spotify_final_dag', 'spotify_etl_final', 'scheduled__2024-06-12T07:10:00+00:00', '--job-id', '188', '--raw', '--subdir', 'DAGS_FOLDER/spotify_final_dag.py', '--cfg-path', '/tmp/tmp47xrjh0i']
[2024-06-12T09:22:17.855+0000] {standard_task_runner.py:91} INFO - Job 188: Subtask spotify_etl_final
[2024-06-12T09:22:18.194+0000] {task_command.py:426} INFO - Running <TaskInstance: spotify_final_dag.spotify_etl_final scheduled__2024-06-12T07:10:00+00:00 [running]> on host d317f063dba7
[2024-06-12T09:22:18.772+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='***@example.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_final_dag' AIRFLOW_CTX_TASK_ID='spotify_etl_final' AIRFLOW_CTX_EXECUTION_DATE='2024-06-12T07:10:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-12T07:10:00+00:00'
[2024-06-12T09:22:18.779+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-12T09:22:18.781+0000] {logging_mixin.py:188} INFO - started
[2024-06-12T09:22:19.659+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-12T09:22:19.660+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spotify_final_dag.py", line 32, in ETL
    df=spotify_etl()
       ^^^^^^^^^^^^^
  File "/opt/airflow/dags/spotify_etl.py", line 83, in spotify_etl
    load_df = return_dataframe()
              ^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spotify_etl.py", line 46, in return_dataframe
    for song in data["items"]:
                ~~~~^^^^^^^^^
KeyError: 'items'
[2024-06-12T09:22:19.728+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=spotify_final_dag, task_id=spotify_etl_final, run_id=scheduled__2024-06-12T07:10:00+00:00, execution_date=20240612T071000, start_date=20240612T092217, end_date=20240612T092219
[2024-06-12T09:22:19.862+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 188 for task spotify_etl_final ('items'; 87)
[2024-06-12T09:22:19.946+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-12T09:22:20.083+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-12T09:22:20.094+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
